src/vs/workbench/contrib/browserView/electron-browser/browserViewActions.ts

```diff
@@ -137,6 +139,14 @@ class ReloadAction extends Action2 {
 	}
 
 	async run(accessor: ServicesAccessor): Promise<void> {
+		const contextKeyService = accessor.get(IContextKeyService);
+		const inCooldownPeriod = contextKeyService.getContextKeyValue(CONTEXT_BROWSER_IN_RELOAD_COOLDOWN_PERIOD.key);
+
+		// No-op if we're in the cooldown period (prevents accidental double-click double reload)
+		if (inCooldownPeriod) {
+			return;
```

Comment:
Can we just do this in the model? i.e. have it optimistically set itself to loading: true / false, and prevent sending the event from there. Then maybe we can remove the timeout too?